WSLGd/FontMonitor.cpp

```diff
@@ -0,0 +1,286 @@
+                if (strcmp(path, USER_DISTRO_FONTPATH) == 0) {
+                    if (std::filesystem::exists(USER_DISTRO_FONTPATH "X11/")) {
+                        AddMonitorFolder(USER_DISTRO_FONTPATH "X11/");
+                    }
+                } else {
+                    // Otherwise, add all existing subfolders to track.
+                    for (auto& dir_entry : std::filesystem::directory_iterator{path}) {
+                        if (dir_entry.is_directory()) {
+                            std::string dirEntry(dir_entry.path().c_str());
+                            dirEntry += "/";
+                            AddMonitorFolder(dirEntry.c_str());
+                        }
+                    }
+                }
+            }
+        } else {
+            LOG_INFO("FontMonitor: %s is already tracked", path);
+        }
+    }
+    CATCH_LOG();
+}
+
+void wslgd::FontMonitor::RemoveMonitorFolder(const char *path)
+{
+    try {
+        std::string monitorPath(path);
+        m_fontMonitorFolders.erase(monitorPath);
+    }
+    CATCH_LOG();
+}
+
+void wslgd::FontMonitor::HandleFolderEvent(struct inotify_event *event)
+{
+    try {
+        std::map<std::string, std::unique_ptr<FontFolder>>::iterator it;
+        for (it = m_fontMonitorFolders.begin(); it != m_fontMonitorFolders.end(); it++) {
+            if (event->wd == it->second->GetWd()) {
+                if (event->mask & (IN_CREATE|IN_MOVED_TO)) {
+                    bool addMonitorFolder = true;
+                    std::string fullPath(it->second->GetPath());
+                    if (fullPath.compare(USER_DISTRO_FONTPATH) == 0) {
+                        /* Immediately under mount folder, only monitor "X11" and its subfolder */
+                        addMonitorFolder = (strcmp(event->name, "X11") == 0);
+                    }
+                    if (addMonitorFolder) {
+                        fullPath += event->name;
+                        fullPath += "/";
+                        AddMonitorFolder(fullPath.c_str());
+                    }
+                } else if (event->mask & (IN_DELETE|IN_MOVED_FROM)) {
+                    std::string fullPath(it->second->GetPath());
+                    fullPath += event->name;
+                    fullPath += "/";
+                    RemoveMonitorFolder(fullPath.c_str());
+                }
+                break;
+            }
+        }
+    }
+    CATCH_LOG();
+}
+
+void wslgd::FontMonitor::HandleFontsDirEvent(struct inotify_event *event)
+{
+    try {
+        std::map<std::string, std::unique_ptr<FontFolder>>::iterator it;
+        for (it = m_fontMonitorFolders.begin(); it != m_fontMonitorFolders.end(); it++) {
+            if (event->wd == it->second->GetWd()) {
+                if (event->mask & (IN_CREATE|IN_CLOSE_WRITE|IN_MOVED_TO)) {
+                    std::string fonts_dir(it->second->GetPath());
+                    fonts_dir += "/";

```

Comment:
fonts_dir could be std::filesystem::path

Reply:
done.